t3f/ops.py

```diff
@@ -20,12 +21,20 @@ def full(tt):
   ranks = shapes.lazy_tt_ranks(tt)
   shape = shapes.lazy_shape(tt)
   raw_shape = shapes.lazy_raw_shape(tt)
+  if isinstance(tt, TensorTrainBatch):
```

Comment:
I don't like in this function the following fact: you mix processing of TensorTrain and TensorTrainBatch up together in the one place.
That's why it's long and a bit hard for understanding.  For example, it's hard to find out a way of a using 'res' variable.
In my opinion, the better way is to make two functions full_tensor_train and full_tensor_train_batch. One of them will be called from full() depends on an input type. This way will make your full() function shorter and better for a future extension with new types.